IteratorsAndGeneratorsTasks.js

```diff
@@ -0,0 +1,116 @@
+//TASK 1//
+let message = 'simple string';
+let messageIterator = message[Symbol.iterator]();
+console.log(messageIterator);
+messageIterator.next();
+for (let item of messageIterator) {
+	console.log(item);
+}
+
+//Task 2//
+
+let getCountdownIterator = function* countdownIterator() {
+	for (let i = 9; i >= 1; i--) {
+		yield i;
+	}
+};
+console.log([...getCountdownIterator()]);
+
+//Task 3//
+
+let todoList = {
+	todoItems: [],
+	addItem(description) {
+		this.todoItems.push({ description, done: false });
+		return this;
+	},
+	crossOutItem(index) {
+		if (index < this.todoItems.length) {
+			this.todoItems[index].done = true;
+		}
+		return this;
+	}
+};
+
+todoList.addItem('task 1').addItem('task 2').addItem('task 3').crossOutItem(0);
+
+let iterableTodoList = {
+	*[Symbol.iterator]() {
+		for (let value = 0; value <= this.todoItems.length - 1; value++) {
```

Comment:
Instead of this : value <= this.todoItems.length - 1
try to use this:  value < this.todoItems.length - this is the most common usage and looks prettier

Reply:
Dont understand how it will work. If i rewrite just with 'value < this.todoItems.length - this' iterator doesn't work)